File: py/level1.py
def answer(n):
    arr = []
    minSum = n
    while minSum:
        last = getMaxSq(minSum)
        minSum -= last
        if last == 0:
            arr.append(1)
        else:
            arr.append(last)
    
    return arr

def getMaxSq(n):
    i = 1
    lastSq = 0
    while 1:
        square = i ** 2
        if square >= n:
            break
        lastSq = square;
        i += 1

    if lastSq == 0:
        lastSq = 1

    return lastSq

File: js/level1.js
console.log('working');

const number = 15324;

console.log(answer(number));

function answer(n) {
    const arr = [];
    let minSum = n;
    while (minSum) {
        const last = getMaxSq(minSum);
        minSum = minSum - last;
        arr.push(last || 1);
    }
    return arr;
}

function getMaxSq (n) {
    let i = 1;
    let lastSq = 0;
    while (true) {
        const mul = Math.pow(i, 2);
        if (mul >= n) {
            break;
        }
        lastSq = mul;
        i++;
    }

    return lastSq || 1;
}